### PR TITLE
WPF support for ARM64: WPF public repo project updates and final source fixes

### PR DESCRIPTION
**Pull request by ryalanms**
WPF support for ARM64: WPF public repo project updates and final source fixes

https://github.com/dotnet/wpf/issues/4041

Co-authored-by: Alexandre Zollinger Chohfi <...>
Co-authored-by: Marcpems <...>
Co-authored-by: Juan Sebastian Hoyos Ayala <...>
Co-authored-by: Ryland <41491307+ryalanms@users.noreply.github.com>

/cc @azchohfi @hoyosjs @marcpems

### CODE REVIEW

File: src/Microsoft.DotNet.Wpf/redist/D3DCompiler/D3DCompiler.vcxproj [hunk 1]
```diff
@@ -24,21 +32,29 @@
...
-  <PropertyGroup>
+  <PropertyGroup Condition="'$(Architecture)'!='arm64'">
     <!-- 
       Target assembly will be a privatized copy of D3DCompiler, like  
       d3dcompiler_47_cor3.dll
     -->
     <TargetName>$(D3DCompilerDllBaseName)$(D3DCompilerVersion)$(WpfVersionSuffix)</TargetName>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Architecture)'=='arm64'">
+    <TargetName>$(D3DCompilerDllBaseName)$(D3DCompilerVersion)</TargetName>
```
Comment (hoyosjs) [inline src/Microsoft.DotNet.Wpf/redist/D3DCompiler/D3DCompiler.vcxproj:43]: Do we need to set this if we are not shipping the component?

File: src/Microsoft.DotNet.Wpf/redist/PresentationNative/PresentationNative.vcxproj [hunk 2]
```diff
@@ -42,41 +48,28 @@
...
-
     <ItemGroup>
       <_PresentationNativeFilePaths Include="@(_PresentationNativeFiles->'%(FilePath)')" />
-      
-      <_PresentationNativeDll Include="@(_PresentationNativeFilePaths)"
-                              Condition="'%(_PresentationNativeFilePaths.Extension)'=='.dll'"/>
-      
-      <_PresentationNativePdb Include="@(_PresentationNativeFilePaths)"
-                              Condition="'%(_PresentationNativeFilePaths.Extension)'=='.pdb'"/>
+      <_PresentationNativeDll Include="@(_PresentationNativeFilePaths)" Condition="'%(_PresentationNativeFilePaths.Extension)'=='.dll'" />
```
Comment (hoyosjs) [inline src/Microsoft.DotNet.Wpf/redist/PresentationNative/PresentationNative.vcxproj:61]: nit: Can you revert the changes in the later half of the file? They were a bit more readable.

File: src/Microsoft.DotNet.Wpf/src/DirectWriteForwarder/DirectWriteForwarder.vcxproj [hunk 3]
```diff
@@ -81,24 +82,21 @@
...
     Ignore ijwhost.dll during packaging
   -->
   <ItemGroup>
     <FileNamesExcludedFromPackaging Include="ijwhost.dll" />
   </ItemGroup>
-
   <Target Name="AddWindowsBaseDefine" AfterTargets="ResolveProjectReferences">
-      <Error Text="Unexpected result from _WindowsBaseReference" Condition="'@(_WindowsBaseReference->Count())' != '1'" />
-      <PropertyGroup>
+    <Error Text="Unexpected result from _WindowsBaseReference" Condition="'@(_WindowsBaseReference-&gt;Count())' != '1'" />
```
Comment (hoyosjs) [inline src/Microsoft.DotNet.Wpf/src/DirectWriteForwarder/DirectWriteForwarder.vcxproj:98]: Was there a reason to change the arrow to an escaped sequence.

File: src/Microsoft.DotNet.Wpf/src/WpfGfx/core/common/utils.cpp [hunk 4]
```diff
@@ -129,20 +129,27 @@ UINT GetPaddedByteCount(UINT cbSize)
     // We want precisely one of _X86_, 
     // _AMD64_ or _ARM_ to be defined. 
     // f(A,B,C) =  A'BC' + AB'C' + A'B'C
+/*
 #if !((!defined(_X86_) && defined(_AMD64_) && !defined(_ARM_)) || \
       (defined(_X86_) && !defined(_AMD64_) && !defined(_ARM_)) || \
-      (!defined(_X86_) && !defined(_AMD64_) && defined(_ARM_)))
+      (!defined(_X86_) && !defined(_AMD64_) && defined(_ARM_));
```
Comment (hoyosjs) [inline src/Microsoft.DotNet.Wpf/src/WpfGfx/core/common/utils.cpp:135]: Was this change intentional? Either we should make the equivalent change or delete this.

File: src/Microsoft.DotNet.Wpf/src/WpfGfx/core/common/utils.cpp [hunk 5]
```diff
@@ -129,20 +129,27 @@ UINT GetPaddedByteCount(UINT cbSize)
...
 #error Exactly one of _X86_, _AMD64_, _ARM_ should be defined
 #endif 
+*/
 
 #if defined(_X86_) 
     return cbSize;
 #elif defined(_AMD64_)
     return cbSize;
-#elif defined(_ARM_)
+// TODO: ARM64PORT: FIX THIS
```
Comment (azchohfi) [inline src/Microsoft.DotNet.Wpf/src/WpfGfx/core/common/utils.cpp:144]: Is this comment stale?

File: src/Microsoft.DotNet.Wpf/src/WpfGfx/core/common/utils.cpp (hunk 4 above)
Comment (ryalanms) [reply src/Microsoft.DotNet.Wpf/src/WpfGfx/core/common/utils.cpp:135]: This macro was commented out.  I've added a case for ARM64 and uncommented it.

File: src/Microsoft.DotNet.Wpf/redist/D3DCompiler/D3DCompiler.vcxproj (hunk 1 above)
Comment (ryalanms) [reply src/Microsoft.DotNet.Wpf/redist/D3DCompiler/D3DCompiler.vcxproj:43]: @azchohfi: Do you have any objections to skipping this entire project for ARM64, or should it be left as-is?

Comment (ryalanms) [reply src/Microsoft.DotNet.Wpf/redist/D3DCompiler/D3DCompiler.vcxproj:43]: I'd prefer to leave this as-is for now, if there are no objections.

**Review by hoyosjs — APPROVED** [review]
